[13.0] session_db

sbidoul opened the pull request:
Same as #2441, for 13.

nseinlet committed:
session_db : store sessions in a database rather than in filestore

sbidoul committed:
session_db: adapt to OCA standards

sbidoul committed:
session_db: do not update write_date on get

The upstream FilesystemSessionStore does not do that.

sbidoul commented:
This one has been used successfully in production since a couple of months now.

sbidoul commented:
This one has been used successfully in production since a couple of months now.

legalsylvain commented on session_db/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1 @@
+Store sessions in a database instead of the filesystem.
```
If possible, could you mention the interest to store sessions in db instead of filesytem ?

sbidoul replied:
@legalsylvain I added this to the README: *This simplifies the configuration of horizontally scalable deployments, by avoiding the need for a distributed filesystem to store the Odoo sessions.*

sbidoul committed:
session_db: explain why such as module is useful

sbidoul commented:
I'm going ahead and merging this. I'll remain available in case any issue show up.

/ocabot merge patch

sbidoul commented:
/ocabot merge patch

sbidoul committed:
session_db: improve resiliency to database errors

Retry on OperationalError exception, which we receive on database restart.
Return cursor to pool when reconnecting.

sbidoul committed:
session_db: fix fallback to original session store

sbidoul commented:
I improved handling of postgres errors. cc/ @adrienpeiffer

sbidoul committed:
session_db: gevent and thread support

There were concurrency issues in evented mode.
So while I was at it, I added support for threaded mode too.

sbidoul committed:
session_db: cosmetics

sbidoul committed:
session_db: improve cursor release

adrienpeiffer reviewed (approved):
Code review and (deep) functional testing

sbidoul commented:
/ocabot merge patch
